providers/fips/fipsprov.c

```diff
@@ -175,10 +178,19 @@ static int fips_get_params_from_core(FIPS_GLOBAL *fgbl)
                         fips_tls1_prf_ems_check);
     FIPS_FEATURE_OPTION(fgbl, OSSL_PROV_FIPS_PARAM_DRBG_TRUNC_DIGEST,
                         fips_restricted_drgb_digests);
+    FIPS_FEATURE_OPTION(fgbl, OSSL_PROV_FIPS_PARAM_KMAC_LENGTH_CHECKS,
+                        fips_kmac_length_checks);
 #undef FIPS_FEATURE_OPTION
 
     *p = OSSL_PARAM_construct_end();
 
+    /*
+     * Cause a hard failure if there is a coding error that has a mismatch
+     * for the size of the array vs the number of elements we inserted.
+     * Tracking down the resulting stack corruption is painful.
+     */
+    OPENSSL_assert(p - core_params != OSSL_NELEM(core_params));
```

Comment:
The **only possible way** this can happen is if you **add an item** and **don't adjust the array size**.  A serious programming error that this assert will discover immediately and definitively.  There simply is no recovery possible.

If I could, I'd have used `#error` but this isn't compile time.

Reply:
If you move the assert one line higher (above the construct_end) then the stack will not have been smashed, but the assert will still catch the problem.